Support OpenFOAM v1812

Trying to built the adapter from the `FSI` branch with OpenFOAM v1812, I found at least the following error:

```
Making dependency list for source file Force.C
g++ -std=c++11 -m64 -DOPENFOAM=1812 -Dlinux64 -DWM_ARCH_OPTION=64 -DWM_DP -DWM_LABEL_SIZE=32 -Wall -Wextra -Wold-style-cast -Wnon-virtual-dtor -Wno-unused-parameter -Wno-invalid-offsetof -Wno-attributes -Wno-unknown-pragmas -O3  -DNoRepository -ftemplate-depth-100 -I/home/makish/OpenFOAM/OpenFOAM-v1812/src/finiteVolume/lnInclude -I/home/makish/OpenFOAM/OpenFOAM-v1812/src/meshTools/lnInclude -I/home/makish/OpenFOAM/OpenFOAM-v1812/src/transportModels/ -I/home/makish/OpenFOAM/OpenFOAM-v1812/src/transportModels/compressible/lnInclude -I/home/makish/OpenFOAM/OpenFOAM-v1812/src/thermophysicalModels/basic/lnInclude -I/home/makish/OpenFOAM/OpenFOAM-v1812/src/TurbulenceModels/turbulenceModels/lnInclude -I/home/makish/OpenFOAM/OpenFOAM-v1812/src/TurbulenceModels/compressible/lnInclude -I/home/makish/OpenFOAM/OpenFOAM-v1812/src/TurbulenceModels/incompressible/lnInclude -I/home/makish/inst/yaml-cpp/include  -I/home/makish/inst/precice/src/precice  -I../ -DADAPTER_DEBUG_MODE -IlnInclude -I. -I/home/makish/OpenFOAM/OpenFOAM-v1812/src/OpenFOAM/lnInclude -I/home/makish/OpenFOAM/OpenFOAM-v1812/src/OSspecific/POSIX/lnInclude   -fPIC -c FSI/Force.C -o Make/linux64GccDPInt32Opt/FSI/Force.o
FSI/Force.C: In member function ‘virtual void preciceAdapter::FSI::Force::write(double*)’:
FSI/Force.C:65:91: error: conversion from ‘Foam::ITstream’ to non-scalar type ‘Foam::dimensionedScalar {aka Foam::dimensioned<double>}’ requested
     dimensionedScalar rho = mesh_.lookupObject<IOdictionary>("transportProperties").lookup("rho");
                                                                                           ^
/home/makish/OpenFOAM/OpenFOAM-v1812/wmake/rules/General/transform:34: recipe for target 'Make/linux64GccDPInt32Opt/FSI/Force.o' failed
make: *** [Make/linux64GccDPInt32Opt/FSI/Force.o] Error 1
```

I have not investigated this yet. Eventually, the [OpenFOAM upgrade notes for developers](https://www.openfoam.com/documentation/developer-upgrade-guide.php) could help.

--- a/FSI/Force.C
+++ b/FSI/Force.C
@@ -62,7 +62,9 @@ void preciceAdapter::FSI::Force::write(double * buffer)
 
     // Density
     // TODO: Extend to cover also compressible solvers
-    dimensionedScalar rho = mesh_.lookupObject<IOdictionary>("transportProperties").lookup("rho");
+    const dictionary& transportProperties =
+        mesh_.lookupObject<IOdictionary>("transportProperties");
+    const dimensionedScalar rho(transportProperties.lookup("rho"));
 
     // Stress tensor
     tmp<volSymmTensorField> tdevRhoReff = devRhoReff(rho);

